### PR TITLE
fix(tooltip): tooltip closes on escape keydown

### LINKED ISSUE #1444: Tooltip: Add ESC key functionality to dismiss
Tooltips should be dismissible with ESC key while focus is on interactive host.

### PR DESCRIPTION
**Pull request by austingardner**
## Description
This PR adds a keydown listener to the tooltip, and then closes the tooltip when the `Escape` key is pressed. 

## References
closes #1444 

## Screenshots
tooltip-escape

### CODE REVIEW

File: src/components/components/ebay-tooltip-base/index.marko [hunk 1]
```diff
@@ -7,12 +7,16 @@ static function toJSON() {
...
 
+static const isBrowser = typeof window !== "undefined";
 $ input.toJSON = toJSON;
 
 <span
     key="container"
     onExpander-expand("handleExpand")
     onExpander-collapse("handleCollapse")
     overlay-style=input.overlayStyle>
+    <!-- <if(isBrowser)>
```
Comment (ianmcburnie) [inline src/components/components/ebay-tooltip-base/index.marko:18]: Are the changes in this file still needed?

Comment (austingardner) [reply src/components/components/ebay-tooltip-base/index.marko:18]: nope I will drop those...

File: src/components/components/ebay-tooltip-base/component-browser.js [hunk 2]
```diff
@@ -10,10 +10,29 @@ module.exports = {
...
+            }
+        }
+    },
+
     onMount() {
         if (this.input.type !== 'dialog--mini') {
             this._setupMakeup();
         }
+
+        this.subscribeTo(window).on('keydown', (e) => {
```
Comment (ianmcburnie) [inline src/components/components/ebay-tooltip-base/component-browser.js:33]: You can/should probably use this instead: https://github.com/eBay/ebayui-core/blob/master/src/common/event-utils/index.js#L32

Example usage here: https://github.com/eBay/ebayui-core/blob/master/src/components/components/ebay-dialog-base/component.js#L37

Comment (ianmcburnie) [reply src/components/components/ebay-tooltip-base/component-browser.js:34]: I think the event delegate should be `this.rootEl` instead of `window`.

Comment (austingardner) [reply src/components/components/ebay-tooltip-base/component-browser.js:34]: So I tried that and it breaks the tooltip... the tooltip doesn't show up at all.

**Comment by agliga** [comment]
Should be ready to go, tooltip dismisses when button is focused and escape key is pressed.

**Comment by austingardner** [comment]
So I don't quite understand why it is better to only close when the tooltip is focused. Isn't it better that the tooltip closes on escape when the user is hovering, as well?

**Comment by agliga** [comment]
> So I don't quite understand why it is better to only close when the tooltip is focused. Isn't it better that the tooltip closes on escape when the user is hovering, as well?

So escape should only be used when the user is using keyboard correct? If they're using mouse, then I don't think escape is needed because they can just mouse out. 
For keyboard use, the only time the tooltip will be shown is on focus.

**Comment by austingardner** [comment]
> > So I don't quite understand why it is better to only close when the tooltip is focused. Isn't it better that the tooltip closes on escape when the user is hovering, as well?
> 
> So escape should only be used when the user is using keyboard correct? If they're using mouse, then I don't think escape is needed because they can just mouse out.
> For keyboard use, the only time the tooltip will be shown is on focus.

Okay that makes sense! For some reason hitting escape just seemed more satisfying and instant than the mouse out. But I agree it isn't needed except on focus.

**Comment by ianmcburnie** [comment]
So, technically you are both right. When I originally filed the issue, the desire was to only close it on ESC when the host had keyboard focus (what Andrew is saying). But given some recent changes to WCAG I can now foresee that we might have to have it close on ESC even if user initiated with mouse hover (what Austin is saying). Please bear with me while I confirm.

**Comment by agliga** [comment]
> So, technically you are both right. When I originally filed the issue, the desire was to only close it on ESC when the host had keyboard focus (what Andrew is saying). But given some recent changes to WCAG I can now foresee that we might have to have it close on ESC even if user initiated with mouse hover (what Austin is saying). Please bear with me while I confirm.

That's fine.
However if we need to have escape on `mouse hover` we need to attach an event to the `window` because we do not know what element is "focused". Another element could be focused while hovering over the tooltip element.

**Comment by ianmcburnie** [comment]
So, as mentioned, this is one of the more recent additions to WCAG: https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html

This one always feels a little odd to me, but basically we need to be able to dismiss the content that appears on mouse hover *without* moving the mouse again. So yep, let's go with the window based event handler that Austin had initially.

**Comment by austingardner** [comment]
> So, as mentioned, this is one of the more recent additions to WCAG: https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html
> 
> This one always feels a little odd to me, but basically we need to be able to dismiss the content that appears on mouse hover _without_ moving the mouse again. So yep, let's go with the window based event handler that Austin had initially.

So why does WCAG concern themselves with content that appears with a mouse? I thought visually impaired users would use the keyboard? Isn't that out of scope for them?

**Comment by austingardner** [comment]
LGTM!

**Review by ianmcburnie — APPROVED** [review]

**Pull request merged**